doc/mgmt/SONiC_PIM_SM_HLD.md

```diff
@@ -0,0 +1,1036 @@
+ 3) Source starts to stream the multicast data
+ 4) FHR DR sends PIM Register message by encapsulating the multicast data and unicast it to RP.
+ 5) RP on receiving PIM Register message, de-encapsulates the messages and and checks if the packet is for an active multicast group, if so it  forwards the packet down the shared tree. RP joins the SPT for the source by sending (S,G) join to FHR so that it can receive native (S,G) multicast traffic. 
+ 6) FHR starts to stream native multicast data to RP. RP on receiving native multicast data sends register stop message to FHR.
+ 7) RP natively forwards the traffic to LHR
+ 8) LHR forwards the muticast stream to the end Host.
+   
+   ![PIM_ASM_SHARED_TREE](pim_images/PIM_SM_Shared_Path_Tree.png "Figure 1: IP Multicast deployment with PIM-SM Shared Tree")
+
+  __Figure 1: IP Multicast Deployment with PIM-ASM Shared Tree__
+
+### 2.1.2 PIM-ASM Shortest Path Tree
+* Shortest path tree also follows the above steps from 1 to 7. The rest all differences are listed below
+8) Shortest Path Tree(SPT) is built from LHR and other transit routers towards the Source. This is achieved by sending an explicit (S,G) join from LHR to source by bypassing the RP. Now LHR has joined the shortest path tree and pull traffic directly from source.
+9) FHR updates the outgoing interface of the (S,G) routes with the interface on which (S,G) join is received from LHR. And thereby it pushes the traffic directly to the LHR.
+10) LHR also prunes the join towards RP for this group by explicitly sending an (S,G,Rpt) prune once the traffic is received on the (S,G) join.
+11) RP on receiving the (S,G,Rpt) prune checks if there are any other interested receivers for this group, if there is no other receiver it sends prune for that group to FHR. And FHR stops streaming the multicast data to RP until any new receiver joins via RP. But, RP maintains the (S,G) state in order to pull the traffic as soon as possbile for any newly joined receiver without the need for registration. To achieve this, FHR periodically sends a NULL register message to RP. Unlike normal PIM register messages, null register messages do not encapsulate the original data packet.NULL registers are sent as long as the source is active.
+    
+    ![PIM_SPT](pim_images/PIM_SM_Shortest_Path_Tree.png "Figure 2: IP Multicast deployment with PIM-ASM Shortest Path Tree")
+ 
+  __Figure 2: IP Multicast Deployment with PIM-ASM Shortest Path Tree__
+
+# 3 Design 
+
+## 3.1 Overview
+
+The following diagram depicts the interactions among modules in SONIC to achieve PIM ASM functionality 
+
+
+  ![PIM_ASM_DESIGN_OVERVIEW](pim_images/PIM_ASM_DESIGN_OVERVIEW.png)
+
+   
+   __Figure 3: PIM ASM Architecture in SONiC__
+
+The interactions between FRR PIM, Linux kernel, ipmcFpmSyncd and ipmcOrch for updating a multicast (*,G) and (S,G) route entry from PIM to the hardware are explained below:
```

Comment:
PIM has (*,G), (S,G) and (S,G) with RPT(known as SG Rpt) . how is this handled? kernel does not have this? So from PIM SG and SG Rpt is treated as separate 2 SG entries if IIF are different? what will happen if IIF is same, Please explain more on this how SGRpt is handled

Reply:
With respect to FRR PIM, there would not be 2 different SG entries.  The SG entries would be with RPF(S) in shortest path tree and SG entry would be with RPF(RP) (SGRpt) in RP tree.